### ISSUE TITLE
How do i Hide Unhide Scene Item which is grouped

### ISSUE THREAD

**Issue by manjit24** [5ony/OBSWebSocketAHK#3]
#NoEnv
SetBatchLines, -1

#Include lib/ObsWebSocket.ahk

class MyOBSController extends ObsWebSocket {

	sceneName := "720X2Canvas"
	sceneItemsByName := {}

	AfterIdentified() {
		this.GetSceneItemList(this.sceneName)
	}

	GetSceneItemListResponse(data) {
		For Key, sceneItemData in data.d.responseData.sceneItems
		{
			this.sceneItemsByName[sceneItemData.sourceName] := {enabled: sceneItemData.sceneItemEnabled, id: sceneItemData.sceneItemId, name: sceneItemData.sourceName}
		}
	}

	toggleSceneItem(sceneItem, visible) {
		if (!this.sceneItemsByName[sceneItem]) {
			return
		}
		this.sceneItemsByName[sceneItem].enabled := visible
		this.SetSceneItemEnabled(this.sceneName, this.sceneItemsByName[sceneItem].id, this.Boolean(this.sceneItemsByName[sceneItem].enabled))
	}

}

obsc := new MyOBSController("ws://127.0.0.1:4455/")

Numpad1::obsc.toggleSceneItem("LeftMic", false)
Numpad2::obsc.toggleSceneItem("LeftMic", true)

![image](https://user-images.githubusercontent.com/40887412/232322569-556b6bfd-74e1-448d-b9b7-43639563d87e.png)

**Comment by 5ony** [comment]
Thanks for your find! Yes, this is an issue, but not really a bug, it is an OBS quirk.

tl;dr: example-toggle-all-scene-elements.ahk is modified to handle groups. I plan to make it easier to use groups as well, but this was the fastest and cleanest thing to do.

longer version: OBS groups are scenes too, but as it is classified as a group, GetGroupSceneItemList() method should be used to request scene items under a named group. When toggling scene items, the scene name should be passed too; in case of a scene item which is under a group, scene name will be the group name (in your example "Group"), even though the actual scene name is different).

Let me know if this solves your issue. Also keep an eye on the repo, I will code the more easier to use version soon.

**Comment by 5ony** [comment]
Introduced a new method to handle all scene items and group items under one scene: GetFullSceneItemList()
Check [example-toggle-all-scene-elements.ahk](https://github.com/5ony/OBSWebSocketAHK/blob/main/example-toggle-all-scene-elements.ahk) for the new example.
Let me know, if I can close this issue.

**Comment by manjit24** [comment]
got it thanks man ... love your work.
